File: Python/pangram.py
def main():
    sentence = [""]
    myInput(sentence)
    output(sentence[0])

def myInput(sentence):
    sentence[0] = input()

def output(sentence):
    print(check(sentence), end='')

def check(sentence):
    letters = {}
    for s in sentence:
        letters[s] = letters.get(s, 0) + 1
    
    isPan = True
    for c in range(ord('a'), ord('z') + 1):
        char = chr(c)
        if letters.get(char, 0) == 0:
            print(f"There is no {char}")
            isPan = False
    return isPan

main()

File: PHP/pangram.php
<?php

function input(){
    return trim(fgets(STDIN));
}

function check($sentence){
    $letters = array();
    foreach (str_split($sentence) as $s){
        $letters[$s] = 1;
    }
    $isPan = true;
    for ($c = ord('a'); $c <= ord('z'); $c++){
        $char = chr($c);
        if (!isset($letters[$char])){
            echo "There is no " . $char . PHP_EOL;
            $isPan = false;
        }
    }
    return $isPan;
}

function output($sentence){
    echo check($sentence) ? 'true' : 'false';
}

$sentence = input();
output($sentence);

?>